lumigator/python/mzai/backend/backend/services/datasets.py

```diff
@@ -168,20 +167,33 @@ def get_dataset_s3_path(self, dataset_id: UUID) -> str:
         return dataset_path
 
     def delete_dataset(self, dataset_id: UUID) -> None:
+        """When the dataset exists, attempts to delete it from both an S3 bucket and the DB.
+
+        Raises exceptions it encounters dealing with S3, except when the file is not found in S3
+        (in this case it deletes the orphaned record from the DB).
+
+        This operation is idempotent, calling it with a record that never existed, or that has
+        already been deleted, will not raise an error.
+        """
         record = self._get_dataset_record(dataset_id)
+        # Early return if the record does not exist (for idempotency).
+        if record is None:
+            return None
 
         try:
-            # Delete from S3
-            # S3 delete is called first => if this fails the DB delete won't take place
+            # S3 delete is called first, if this fails for any other reason that the file not being
+            # found, then the DB delete won't take place, and an exception raised.
             dataset_key = self._get_s3_key(record.id, record.filename)
             dataset_path = f"s3://{ Path(settings.S3_BUCKET) / dataset_key }"
             self.s3_filesystem.rm(dataset_path, recursive=True)
+        except FileNotFoundError as e:
+            # File not found errors are allowed, but we perform clean-up in this situation.
+            logger.error(f"Dataset ID: {dataset_id} was present in the DB but not found on S3... "
+                         f"Cleaning up DB by removing ID. {e}")
```

Comment:
being this a trapped exception for an error we knew did not give us useful information, do we need to get and print `e` here?

Reply:
I was thinking it would be useful to emit everything we had into the logs in case someone wants to go looking around for more information. 

Happy to drop/swallow it if you think it's not worthwhile 😄

Reply:
I remember s3fs was quite minimal in the info it returned, but I think it's worth keeping it if we see, after testing, that there's anything useful in there

Reply:
After pairing up on this we agreed to leave the exception in the log as it may be useful in diagnosing any discrepancies between S3 paths etc.